packages/richtext-lexical/babel.config.cjs

```diff
@@ -0,0 +1,36 @@
+const fs = require('fs')
+
+const ReactCompilerConfig = {
+  sources: (filename) => {
+    if (!filename.endsWith('.tsx') && !filename.endsWith('.jsx') && !filename.endsWith('.js')) {
+      return false
+    }
+
+    // read file and check if 'use client' is at top. if not, return false
+    // if it is, return true
+    const file = fs.readFileSync(filename, 'utf8')
+    if (file.includes("'use client'")) {
+      //console.log("Compiling: " + filename)
+      return true
+    }
+    console.log('Skipping: ' + filename)
+    return false
+  },
+  //runtimeModule: "react"
+}
```

Comment:
I'd suggest also skipping compiling node_modules, we've seen that there can be surprising behavior there because React Compiler was designed to run on source code and not pre-compiled/minified/etc code that one typically sees in node_modules